examples/templates/java/README.md

```diff
@@ -0,0 +1,252 @@
+
+## Local execution
+To execute this pipeline locally, specify the parameters:
+- Kafka Bootstrap servers
+- Kafka input topics
+- Pub/Sub output topic
+in the following format:
+```bash
+--bootstrapServers=host:port \
+--inputTopics=your-input-topic \
+--outputTopic=projects/your-project-id/topics/your-topic-pame
+```
+Optionally, to retrieve Kafka credentials for SASL/SCRAM,
+specify a URL to the credentials in HashiCorp Vault and the vault access token:
+```bash
+--secretStoreUrl=http(s)://host:port/path/to/credentials
+--vaultToken=your-token
+```
+Optionally, to configure secure SSL connection between the Beam pipeline and Kafka,
+specify the parameters:
+- A local path to a truststore file
+- A local path to a keystore file
+- Truststore password
+- Keystore password
+- Key password
+```bash
+--truststorePath=path/to/kafka.truststore.jks
+--keystorePath=path/to/kafka.keystore.jks
+--truststorePassword=your-truststore-password
+--keystorePassword=your-keystore-password
+--keyPassword=your-key-password
+```
+To change the runner, specify:
+```bash
+--runner=YOUR_SELECTED_RUNNER
+```
+See examples/java/README.md for steps and examples to configure different runners.
+
+## Google Dataflow Template
+
+### Setting Up Project Environment
+
+#### Pipeline variables:
+
+```
+PROJECT=id-of-my-project
+BUCKET_NAME=my-bucket
+REGION=my-region
+```
+
+#### Template Metadata Storage Bucket Creation
+
+The Dataflow Flex template has to store its metadata in a bucket in
+[Google Cloud Storage](https://cloud.google.com/storage), so it can be executed from the Google Cloud Platform.
+Create the bucket in Google Cloud Storage if it doesn't exist yet:
+
+```
+gsutil mb gs://${BUCKET_NAME}
+```
+
+#### Containerization variables:
+
+```
+IMAGE_NAME=my-image-name
+TARGET_GCR_IMAGE=gcr.io/${PROJECT}/${IMAGE_NAME}
+BASE_CONTAINER_IMAGE=my-base-container-image
+TEMPLATE_PATH="gs://${BUCKET_NAME}/templates/kafka-pubsub.json"
+```
+
+### Creating the Dataflow Flex Template
+
+Dataflow Flex Templates package the pipeline as a Docker image and stage these images
+on your project's [Container Registry](https://cloud.google.com/container-registry).
+
+To execute the template you need to create the template spec file containing all
+the necessary information to run the job. This template already has the following
+[metadata file](kafka-to-pubsub/src/main/resources/kafka_to_pubsub_metadata.json) in resources.
+
+Navigate to the template folder:
+
+```
+cd /path/to/beam/examples/templates/java/kafka-to-pubsub
+```
+
+Build the Dataflow Flex Template:
+
+```
+gcloud dataflow flex-template build ${TEMPLATE_PATH} \
+       --image-gcr-path ${TARGET_GCR_IMAGE} \
+       --sdk-language "JAVA" \
+       --flex-template-base-image ${BASE_CONTAINER_IMAGE} \
+       --metadata-file "src/main/resources/kafka_to_pubsub_metadata.json" \
+       --jar "build/libs/beam-examples-templates-java-kafka-to-pubsub-2.25.0-SNAPSHOT-all.jar" \
```

Comment:
nit: this shouldn't specify a specific version
```suggestion
       --jar "build/libs/beam-examples-templates-java-kafka-to-pubsub-<version>-all.jar" \
```

Reply:
fixed